spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/ParquetReaderType.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark;
+
+/** Enumerates the types of Parquet readers. */
+public enum ParquetReaderType {
+  /** ICEBERG type utilizes the Parquet reader from Apache Iceberg. */
+  ICEBERG,
+
+  /**
+   * COMET type changes the Parquet reader to the Apache DataFusion Comet Parquet reader. Comet
+   * Parquet reader performs I/O and decompression in the JVM but decodes in native to improve
+   * performance. Additionally, Comet will convert Spark's physical plan into a native physical plan
+   * and execute this plan natively.
+   *
+   * <p>TODO: Implement {@link org.apache.comet.parquet.SupportsComet} in SparkScan to convert Spark
```

Comment:
It seems `SupportsComet` is an empty marker interface. Does Comet native execution has a dependency on Iceberg code? How does it perform the conversion?

Reply:
Comet checks [SupportsComet. isCometEnabled()](https://github.com/apache/datafusion-comet/blob/main/spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala#L163) and wraps BatchScanExec with CometBatchScanExec if isCometEnabled is true. I will make `SparkScan` implement `SupportsComet` and return true for `isCometEnabled`.

Reply:
So this marker interface avoid the need to depend on Iceberg classes? Okay, that makes sense.

Reply:
Do we have to include this TODO here, however? It doesn't seem to belong here.

Reply:
I want to have a TODO note somewhere so that people will know that native execution is not yet supported. There are some additional steps we need to take to make native execution work. Otherwise people may think native execution is enabled by this PR. Let me see if there is a better place for this.